src/ttsim/interface_dag_elements/data_converters.py

```diff
@@ -131,12 +132,19 @@ def df_with_mapped_columns_to_flat_data(
 
 
     """
+    # Always use numpy for initial data preparation (performance optimization for JAX)
     path_to_array = {}
     for path, mapper_value in dt.flatten_to_tree_paths(mapper).items():
-        if xnp.isscalar(mapper_value) and not isinstance(mapper_value, str):
-            path_to_array[path] = xnp.asarray([mapper_value] * len(df))
+        if np.isscalar(mapper_value) and not isinstance(mapper_value, str):
+            numpy_array = np.asarray([mapper_value] * len(df))
         else:
-            path_to_array[path] = xnp.asarray(df[mapper_value])
+            numpy_array = np.asarray(df[mapper_value])
+
+        # Convert to target backend array with zero-copy if JAX
+        if xnp.__name__ == "jax.numpy":
```

Comment:
```suggestion
        if backend == "jax":
```
Just add `backend` as an argument to the function.